File: 1. 단계별로_풀어보기/10~19/12. 브루트 포스/1. 2798.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N, M = map(int, input().split())
cardArr = list(map(int, input().split()))

cardArr.sort()

maxSum = 0

for i in range(N):
  card0 = cardArr[i]

  for j in range(i + 1, N):
    card1 = cardArr[j]
    sum01 = card0 + card1

    # 혹여나 중간에 합이 넘어버리면 탈출
    if sum01 >= M:
      break

    for k in range(j + 1, N):
      sum012 = sum01 + cardArr[k]

      if sum012 <= M:
        maxSum = max(maxSum, sum012)
      else: # 여기도 마찬가지로 합이 넘어버리면 반복문을 더 돌 필요가 없다
        break

output(f"{maxSum}")

File: 1. 단계별로_풀어보기/10~19/12. 브루트 포스/1. 2798.js
// 입력
const inputArr = require('fs').readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const [N, M] = inputArr[0].split(' ').map(Number)
const cardArr = inputArr[1].split(' ').map(Number)

// 결과값 담을 변수
let result = 0

// 연산
for (let i = 0; i < N; i++) {
  const card0 = cardArr[i] // 성능 최적화: 메모리 참조 연산 줄이기

  for (let j = i + 1; j < N; j++) {
    const card1 = cardArr[j] // 성능 최적화: 메모리 참조 연산 줄이기

    for (let k = j + 1; k < N; k++) {
      const sumCard = card0 + card1 + cardArr[k] // 가독성 위한 변수 선언

      if (sumCard <= M) {
        result = Math.max(result, sumCard)
      }
    }
  }
}

// 출력
process.stdout.write(`${result}`)
